src/Fantomas.Core/CodePrinter.fs

```diff
@@ -2058,9 +2072,10 @@ let genControlExpressionStartCore
 let genMultilineInfixExpr (node: ExprInfixAppNode) =
     let genLhs (ctx: Context) =
         match node.LeftHandSide with
-        | Expr.IfThen _
-        | Expr.IfThenElse _
-        | Expr.IfThenElif _ when (ctx.Config.IndentSize - 1 <= node.Operator.Text.Length) ->
+        | _ when
```

Comment:
Should be a partial active pattern. It is ok to have the when clause for `when (ctx.Config.IndentSize - 1 <= node.Operator.Text.Length)`.
Maybe we even want that as a separate let binding to keep the `when` short.